### ISSUE TITLE
[Question] How to use ForeignKeySchema in migration?

### ISSUE THREAD

**Issue by dinfer** [typeorm/typeorm#792]
I'm using migration to create a table which has a foreign key. But there is no example. I tried to find something in the source, the comment didn't explain anything meanful.

```
constructor(name: string, columnNames: string[], referencedColumnNames: string[], referencedTable: string, onDelete?: string);
```

The `columnNames` and the `referecedColumnNames` have the **same** comment, It confused me a lot.(see [here](https://github.com/typeorm/typeorm/blob/1a2c9e091c56a22b4d33a8071802c52a34159aee/src/schema-builder/schema/ForeignKeySchema.ts#L17-L20) for `columnNames` and [here](https://github.com/typeorm/typeorm/blob/1a2c9e091c56a22b4d33a8071802c52a34159aee/src/schema-builder/schema/ForeignKeySchema.ts#L27-L30) for `referecedColumnNames`), how do I use them? What's more, I found a `todo`(see [here](https://github.com/typeorm/typeorm/blob/13fb2fc8d989ce58c7402acc608c40085dfb12fa/src/driver/oracle/OracleQueryRunner.ts#L383)), Does `ForeignKeySchema` works fine?

My code snippet

```
    public async up(queryRunner: QueryRunner): Promise<any> {
        await queryRunner.createTable(new TableSchema('account', [
            new ColumnSchema({ name: 'id', type: 'int',
                isPrimary: true, isGenerated: true, generationStrategy: 'increment'}),
            new ColumnSchema({ name: 'uid', type: 'varchar', length: 100,
                isNullable: false}),
            new ColumnSchema({ name: 'balance', type: 'decimal(20, 2)',
                default: 0}),
        ]))
        await queryRunner.createTable(new TableSchema('transfer', [
            new ColumnSchema({ name: 'id', type: 'int',
                isPrimary: true, isGenerated: true, generationStrategy: 'increment'}),
            new ForeignKeySchema('fromAccountId', ['fromAccountId'], ['id'], 'account'),
            // some other fields
        ]))
    }
```

The generated SQL code that cannot work

```
CREATE TABLE `transfer` (
	`id` INT (11) NOT NULL PRIMARY KEY AUTO_INCREMENT COMMENT '主键',
	`fromAccountId` NOT NULL,
        # some other fields
)
```

**Comment by pleerock** [comment]
`ForeignKeySchema` isn't finished public API yet and for now I recommend to use raw sql queries in your migrations instead if you have issues with it

**Comment by AlexMesser** [comment]
fixed and released in `typeorm@0.2.0-alpha.23`. You can try it via `npm i typeorm@next`.
